File: nature-harvest-frontend/src/app/responses/order/order-tracking.response.ts
import { OrderAndOrderDetailsResponse } from './order-orderdetails-response';

export class OrderTrackingResponse {
  orders: OrderAndOrderDetailsResponse[];
  totalPages: number;
  totalPending: number;
  totalConfirmed: number;
  totalPickedUp: number;
  totalOnTheWay: number;
  totalSuccessDelivery: number;
  totalRefused: number;
  totalCanceled: number;
  constructor(data: any) {
    this.orders = data.orders;
    this.totalPages = data.totalPages;
    this.totalPending = data.totalPending;
    this.totalConfirmed = data.totalConfirmed;
    this.totalPickedUp = data.totalPickedUp;
    this.totalOnTheWay = data.totalOnTheWay;
    this.totalSuccessDelivery = data.totalSuccessDelivery;
    this.totalRefused = data.totalRefused;
    this.totalCanceled = data.totalCanceled;
  }
}

File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/responses/order/OrderTrackingResponse.java
package com.api.nature_harvest_backend.responses.order;

import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderTrackingResponse {
    private List<OrderAndOrderDetailsResponse> orders;
    private int totalPages;
    private long totalPending;
    private long totalConfirmed;
    private long totalPickedUp;
    private long totalOnTheWay;
    private long totalSuccessDelivery;
    private long totalRefused;
    private long totalCanceled;
}
